pkg/resource/secret/sdk.go

```diff
@@ -0,0 +1,549 @@
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package secret
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackcondition "github.com/aws-controllers-k8s/runtime/pkg/condition"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/secretsmanager"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	svcapitypes "github.com/aws-controllers-k8s/secretsmanager-controller/apis/v1alpha1"
+)
+
+// Hack to avoid import errors during build...
+var (
+	_ = &metav1.Time{}
+	_ = strings.ToLower("")
+	_ = &aws.JSONValue{}
+	_ = &svcsdk.SecretsManager{}
+	_ = &svcapitypes.Secret{}
+	_ = ackv1alpha1.AWSAccountID("")
+	_ = &ackerr.NotFound
+	_ = &ackcondition.NotManagedMessage
+	_ = &reflect.Value{}
+	_ = fmt.Sprintf("")
+	_ = &ackrequeue.NoRequeue{}
+)
+
+// sdkFind returns SDK-specific information about a supplied resource
+func (rm *resourceManager) sdkFind(
+	ctx context.Context,
+	r *resource,
+) (latest *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkFind")
+	defer func() {
+		exit(err)
+	}()
+	// If any required fields in the input shape are missing, AWS resource is
+	// not created yet. Return NotFound here to indicate to callers that the
+	// resource isn't yet created.
+	if rm.requiredFieldsMissingFromReadOneInput(r) {
+		return nil, ackerr.NotFound
+	}
+
+	input, err := rm.newDescribeRequestPayload(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *svcsdk.DescribeSecretOutput
+	resp, err = rm.sdkapi.DescribeSecretWithContext(ctx, input)
+	rm.metrics.RecordAPICall("READ_ONE", "DescribeSecret", err)
+	if err != nil {
+		if reqErr, ok := ackerr.AWSRequestFailure(err); ok && reqErr.StatusCode() == 404 {
+			return nil, ackerr.NotFound
+		}
+		if awsErr, ok := ackerr.AWSError(err); ok && awsErr.Code() == "ResourceNotFoundException" {
+			return nil, ackerr.NotFound
+		}
+		return nil, err
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := r.ko.DeepCopy()
+
+	if ko.Status.ACKResourceMetadata == nil {
```

Comment:
`ko.Status.ID` is not set anywhere below, which will cause the controller to infinitely try to create a secret because of L67

Reply:
add hook in sdkFind and sdkCreate to set ID
```go
	if resp.ARN != nil {
		ko.Status.ID = resp.ARN
	}
```